### ISSUE TITLE
Always enable MBEDTLS_PK_USE_PSA_EC_DATA

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#9676]
Always enable the internal option `MBEDTLS_PK_USE_PSA_EC_DATA`, and remove the code that was gated by `!defined(MBEDTLS_PK_USE_PSA_EC_DATA)`.

We want this because we want less adherence to legacy crypto APIs. It'll make the [removal of `!MBEDTLS_USE_PSA_CRYPTO`](https://github.com/Mbed-TLS/mbedtls/issues/9631) easier.

**Comment by gilles-peskine-arm** [comment]
https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/204 turns the always option on, but that's a small patch in terms of code changes and can be done independently.

**Comment by gilles-peskine-arm** [comment]
Without digging down, I would guess that we were only using these functions when TLS could accessed underlying key representations directly, i.e. when `MBEDTLS_USE_PSA_CRYPTO` was disabled. In any case, they aren't part of a public API, so if they aren't used internally, please remove them.

**Comment by valeriosetti** [comment]
> Without digging down, I would guess that we were only using these functions when TLS could accessed underlying key representations directly, i.e. when `MBEDTLS_USE_PSA_CRYPTO` was disabled. In any case, they aren't part of a public API, so if they aren't used internally, please remove them.

In order to keep PR small and with a clear focus, I created a specific issue for this --> https://github.com/Mbed-TLS/mbedtls/issues/10164

**Comment by gilles-peskine-arm** [comment]
Ah, yes, I had forgotten when I created this issue that `MBEDTLS_PK_USE_PSA_EC_DATA` is incompatible with restartable ECC. `MBEDTLS_ECP_RESTARTABLE` is not deprecated, we want it to keep working.

The medium-term solution is to make the code in PK that's specifically about restartable behavior switch from calling `mbedtls_ecdsa_xxx_restartable` to `psa_xxx_hash_interruptible_yyy`. I haven't estimated how big a task this is. We should try to determine whether it's less work to do it now, or to keep coping with the non-PSA code paths for when restartable ECC is enabled.

**Comment by mpg** [comment]
@valeriosetti Could you give the `psa_xxx_hash_interruptible` way a quick try, just enough to gauge if it would work easily or be a lot of work? If that works nicely, that would be the best outcome, because that's what we want in the end. If it turns out that's more work that we can realistically absorb now, we'll probably have to keep `MBEDTLS_PK_USE_PSA_EC_DATA` depending on `!MBEDTLS_ECP_RESTARTABLE` for now (which would be a pain).

**Comment by valeriosetti** [comment]
> Could you give the `psa_xxx_hash_interruptible` way a quick try, just enough to gauge if it would work easily or be a lot of work?

Sure! If that works then I will file up a new issue which will be a prerequisite for this one.

**Comment by valeriosetti** [comment]
For the record I also created https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/283 to track the removal of USE_PSA_CRYPTO from the PK module
